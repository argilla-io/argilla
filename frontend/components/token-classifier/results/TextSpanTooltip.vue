<template>
  <span class="highlight__label">
    <span class="highlight__tooltip__container">
      <span class="highlight__tooltip">
        <span class="highlight__tooltip__origin" v-if="span.origin">{{
          span.origin === "prediction" ? "pred." : "annot."
        }}</span>
        <span>{{ span.entity.label }} </span>
      </span>
    </span>
  </span>
</template>

<script>
export default {
  props: {
    span: {
      type: Object,
      required: true,
    },
  },
};
</script>

<style lang="scss" scoped>
.highlight {
  &__label {
    min-height: 40px;
    @include font-size(16px);
    line-height: 20px;
  }
  &__tooltip {
    cursor: default;
    display: block;
    border-radius: 2px;
    padding: 5px 10px 6px 10px;
    margin-bottom: 0.5em;
    transition: opacity 0.5s ease, z-index 0.2s ease;
    white-space: nowrap;
    user-select: none;
    font-weight: 600;
    min-width: 80px;
    @include font-size(16px);
    .prediction & {
      margin-top: 0.5em;
    }
    & > span {
      display: block;
    }
    &__container {
      position: absolute;
      right: 50%;
      transform: translateX(50%);
      opacity: 1;
      transition-delay: 0s;
      z-index: 4;
      .annotation & {
        top: -77px;
      }
      .prediction & {
        top: 13px;
      }
    }
    &__origin {
      @include font-size(12px);
      font-weight: normal;
    }
  }
  &__tooltip:after {
    margin: auto;
    position: absolute;
    right: 0;
    left: 0;
    .annotation & {
      @include triangle(bottom, 6px, 6px, auto);
      bottom: 2px;
    }
    .prediction & {
      @include triangle(top, 6px, 6px, auto);
      top: 3px;
    }
  }
}
</style>
