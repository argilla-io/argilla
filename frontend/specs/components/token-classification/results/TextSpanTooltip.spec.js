import { mount } from "@vue/test-utils";
import TextSpanTooltip from "@/components/token-classifier/results/TextSpanTooltip";

function mountTextSpanTooltip() {
  return mount(TextSpanTooltip, {
    propsData: {
      span: {
        end: 73,
        entity: {
          end: 73,
          end_token: 15,
          label: "Arreglo",
          start: 57,
          start_token: 13,
        },
        origin: "annotation",
      },
    },
  });
}

describe("TextSpanTooltip", () => {
  let spy = jest.spyOn(console, "error");
  afterEach(() => spy.mockReset());

  test("renders properly", () => {
    const wrapper = mountTextSpanTooltip();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
