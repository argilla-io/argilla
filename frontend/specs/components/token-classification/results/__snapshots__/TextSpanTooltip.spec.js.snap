// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSpanTooltip renders properly 1`] = `<span class="highlight__label"><span class="highlight__tooltip__container"><span class="highlight__tooltip"><span class="highlight__tooltip__origin">annot.</span> <span>Arreglo </span></span></span></span>`;
